populus/utils/hexadecimal.py

```diff
@@ -0,0 +1,8 @@
+from hexbytes import HexBytes
+
+
+def hexbytes_to_hexstr(val):
```

Comment:
Hey @kkom here! :) I came across a problem with this function when trying to compile a Vyper contract.

The function makes the assumption that `val` can be of either type `HexBytes` or `str`, which unfortunately isn't true.

More discussion (and a proposed alternative) available where I commented about this issue first: https://github.com/voith/populus/commit/37139e5d0dce609cc6971e603da3cc1a5554701b#r30091655

Reply:
@kkom Thanks for reporting. This is one portion that I forgot to test manually. Let me add a

Reply:
My local patch is good enough, as it lets me compile code (and hopefully deploy? Haven’t tested it yet.). I’m going to use some other framework for unit and integration testing in the meantime.

I’ll wait for you to merge your commits into master, and then dig around to see what I can fix. :)

Let me know if you need any help, with PR review, etc

Reply:
@kkom https://github.com/ethereum/populus/pull/474/commits/032eb453bce753afaae5b526db5eb18d02c28c3c + https://github.com/ethereum/populus/pull/475 Should fix the problem.

I'm just waiting for someone to review #474.

Reply:
#474 got merged!